app/controllers/api/chat_channels_controller.rb

```diff
@@ -13,13 +13,23 @@ def update
     end
 
     editable_params = params.permit(*CHAT_CHANNEL_EDITABLE_PARAMS)
+    editable_params.delete(:auto_join_users) if !chat_channel.category_channel?
     editable_params.each do |k, v|
       editable_params[k] = nil if editable_params[k].blank?
     end
+
+    if ActiveRecord::Type::Boolean.new.deserialize(editable_params[:auto_join_users])
```

Comment:
do we actually need to check it's a valid boolean here ? We might just check you attempted to change auto_join_users ?

Reply:
We only want to perform the rate limiting when enabling the option because we trigger the auto-join then. When setting it to false, it's just a regular update, so it should be ok to bypass the rate limiter.